DisCatSharp/EventAttributes.cs

```diff
@@ -0,0 +1,51 @@
+// This file is part of the DisCatSharp project, based off DSharpPlus.
+//
+// Copyright (c) 2021-2022 AITSYS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#nullable enable
+
+using System;
+
+namespace DisCatSharp
+{
+	/// <summary>
+	/// Methods marked with this attribute will be registered as event handling methods
+	/// if the associated type / an associated instance is being registered.
+	/// </summary>
+	[AttributeUsage(AttributeTargets.Method)]
+	public class Event : Attribute
```

Comment:
Attributes have the `Attribute` suffix, this helps identify them. Luckily, C# tooling is great and this doesn't impact how you implement it on the target method. Can still use it as `[Event]` --- but when trying to grab it via reflection you'd be grabbing `EventAttribute`

Reply:
I didn't know this, cool!